typescript/cli/src/deploy/core.ts

```diff
@@ -127,7 +127,11 @@ export async function runCoreApply(params: ApplyParams) {
   if (transactions.length) {
     logGray('Updating deployed core contracts');
     for (const transaction of transactions) {
-      await multiProvider.sendTransaction(chain, transaction);
+      await multiProvider.sendTransaction(
+        // Using the provided chain id because there might be remote chain transactions included in the batch
+        transaction.chainId ?? chain,
+        transaction,
```

Comment:
We use the provided chain id in the transaction because using the `chain` variable by default assumed that all txs were done on the same chain which is not the case anymore now that we also send txs on remote chains to register the current router.

Reply:
but then inside the multiprovider it would get the chain for that chainid, thinking it is the domainid

which affects chains such as rootstock/immutable/flow where we have new domainids that are different to the chainids. I think so far it hasn't broken anything because it's basically the same chain metadata with different domain IDs - but not to say it doesn't break in the future 

also when we get the update txs here, we're assuming that the current signer is the owner - which might not be the case for SAFE/ICA owned chains. I think the update txs should go through a tx submitter instead of the raw multiprovider?

Reply:
> which affects chains such as rootstock/immutable/flow where we have new domainids that are different to the chainids. I think so far it hasn't broken anything because it's basically the same chain metadata with different domain IDs - but not to say it doesn't break in the future

Ok so probably the chain id field on the annotated tx should be updated to `domainId` or something similar so to remove the confusion there too no?  

> also when we get the update txs here, we're assuming that the current signer is the owner - which might not be the case for SAFE/ICA owned chains. I think the update txs should go through a tx submitter instead of the raw multiprovider?

yep at the moment support for the submitter is only implemented on `warp apply` but will be added to `core apply` too. I can do that in a follow up pr when I add ICA support to `warp apply` after #4930 is merged

Reply:
> Ok so probably the chain id field on the annotated tx should be updated to domainId or something similar so to remove the confusion there too no?

this is why I wanted `chain` on that type back in https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4798 but then the diff would be quite big so we opted to minimise the diff